### PR TITLE
CNAME-compatible way of using _dnslink as per 0.4

### PR DESCRIPTION
**Pull request by dokterbob**
Document use of CNAME for hosting ipfs sites.
Ref: https://ipfs.io/blog/14-ipfs-0-4-0-released/

### CODE REVIEW

File: examples/websites/README.md [hunk 1]
```diff
@@ -45,12 +45,12 @@ Published to <your peer id>: /ipfs/$SITE_HASH
...
 
-So, if you point `your.domain`'s A record to the IP of `gateway.ipfs.io`, and then wait for the DNS to propogate, then anyone should be able to access your ipfs-hosted site without any extra configuration simply at `http://your.domain`.
+So, if you create a CNAME for `your.domain` pointing to `gateway.ipfs.io`, and then wait for the DNS to propogate, then anyone should be able to access your ipfs-hosted site without any extra configuration simply at `http://your.domain`.
```
Comment (jbenet) [inline examples/websites/README.md]: Should have both options i think, and describe the tradeoffs. cc @lgierth

Comment (dokterbob) [inline examples/websites/README.md]: What _are_ the tradeoffs, specifically, the disadvantages of `_dnslink` entries over the domain itself?

Giving people multiple options is also a potential source of confusion, especially in examples (where often 'best practises' seem to be demonstrated). Although there is also a clear case for documenting, as reference, the full functionality of the software.

Comment (ghost) [reply examples/websites/README.md]: The only difference is that `_dnslink.` allows you to use a CNAME record, which is useful to quickly point a new domain name at an IPFS object and gateway:

```
CNAME example.com => gateway.ipfs.io.
TXT _dnslink.example.com => dnslink=/ipfs/<hash>
```

It's not possible to mix CNAME with other record types, which is why you can't just have a `CNAME example.com` record in the above example.

Also note how the CNAME record is pointing to gateway.ipfs.io, not ipfs.io -- we don't want to apply subdomains and MX records!

**Comment by dokterbob** [comment]
I've described `_dnslink` as an equal alternative. Let me know what you think (or just merge hehe).

**Comment by Kubuxu** [comment]
LGTM, @lgierth do you also want to take a look?

**Comment by ghost** [comment]
Thanks, this LGTM now :+1:

**Pull request merged**